pkg/services/publicdashboards/service/service.go

```diff
@@ -215,6 +215,11 @@ func (pd *PublicDashboardServiceImpl) Update(ctx context.Context, u *user.Signed
 		return nil, ErrPublicDashboardNotFound.Errorf("Update: public dashboard not found by uid: %s", dto.Uid)
 	}
 
+	// validate the public dashboard belongs to the dashboard
+	if existingPubdash.DashboardUid != dto.DashboardUid {
+		return nil, ErrInvalidUid.Errorf("Update: the public dashboard does not belong to the dashboard")
+	}
+
```

Comment:
shouldn't we make the update with a WHERE claused based on the dashboard uid? 
`WHERE dashboard_uid = cmd.PublicDashboard.DashboardUid`
It's a client problem if the dashboard uid doesn't fit correctly. 
And maybe if the updated record is 0, throw this error?

Reply:
We could but if other service uses the `store.Update` func it should provide the `dashboardUid` as well, it's an unnecessary dependency. The `publicDashboardUid` is the PK, it's enough to identify a public dashboard so from a layer perspective it's ok to use only the `uid` in the Update. The problem is we require the `dashboardUid` in the URL and we were not using it for validation.

Reply:
You're right. It's weird to have the dashboardUid in the url if you don't really need it and if publicDashboardUid is the one that uniquely identifies the dashboard